ee/control/consumers/flareconsumer/flareconsumer.go

```diff
@@ -0,0 +1,59 @@
+package flareconsumer
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/debug/shipper"
+)
+
+const (
+	// Identifier for this consumer.
+	FlareSubsystem = "flare"
+)
+
+type FlareConsumer struct {
+	flarer   flarer
+	knapsack types.Knapsack
+	// newFlareStream is assigned to a field so it can be mocked in tests
+	newFlareStream func(note, uploadURL string) (io.WriteCloser, error)
+}
+
+type flarer interface {
+	RunFlare(ctx context.Context, k types.Knapsack, flareStream io.WriteCloser) error
+}
+
+func New(knapsack types.Knapsack) *FlareConsumer {
+	return &FlareConsumer{
+		flarer:   &FlareRunner{},
+		knapsack: knapsack,
+		newFlareStream: func(note, uploadURL string) (io.WriteCloser, error) {
+			return shipper.New(knapsack, shipper.WithNote(note), shipper.WithUploadURL(uploadURL))
+		},
+	}
+}
+
+func (fc *FlareConsumer) Do(data io.Reader) error {
+	if fc.flarer == nil {
+		return errors.New("flarer is nil")
+	}
+
+	flareData := struct {
+		Note      string `json:"note"`
```

Comment:
I'm a little confused about why we need `Note` here. I know why we need it on the launcher initiated ones, but on the server initiated ones I'm less sure.

Reply:
good point, dropped from consumer